File: hw/Pérez/U1/HW09ChickenFarmSimulatorjs/ChickenFarmSimulator.js
/**
 * @autor Carlos Pérez, The Java Bandits, DCCO-ESPE
 */
class ChickenFarmSimulator {
    static main(args) {
        console.log("Hello chickens from Carlos Perez");
        console.log("Chicken Farm Simulator 2K24 V 0.1");

        let chicken = new Chicken(1, "Lucy", "Black", 1, true, new Date());

        console.log("Chicken -->", chicken);

        let id = 2;
        let name = "Maruja";
        let color = "white";
        let age = 0;
        let molting = false;
        let bornOnDate = new Date();

        chicken = new Chicken(id, name, color, age, molting, bornOnDate);
        console.log("Chicken -->", chicken);
    }
}

class Chicken {
    constructor(id, name, color, age, molting, bornOnDate) {
        this.id = id;
        this.name = name;
        this.color = color;
        this.age = age;
        this.molting = molting;
        this.bornOnDate = bornOnDate;
    }

    toString() {
        return `Chicken{id=${this.id}, name='${this.name}', color='${this.color}', age=${this.age}, molting=${this.molting}, bornOnDate=${this.bornOnDate}}`;
    }
}

// Ejecutar la función principal
ChickenFarmSimulator.main();

File: ws/travez/u1/WS11Collections/src/ec/edu/espe/farmsimulator/view/ChickenFarmSimulator.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.farmsimulator.view;

import UtilityPackages.FileManager;
import ec.edu.espe.farmsimulator.model.Chicken;
import java.util.Scanner;
import java.time.LocalDate;
import java.time.format.DateTimeFormatter;

/**
 *
 * @author Ismael Travez, The Java Bandits, DCCO-ESPE
 *
 */
public class ChickenFarmSimulator {

    public static void main(String[] args) {

        DateTimeFormatter dateFormat = DateTimeFormatter.ofPattern("dd/MM/yyyy");

        System.out.println("Hello chickens from Imsale Travez \nChicken Farm Simulator 2K24 V 0.4   ");

        Chicken chicken = new Chicken(1, "Lucy", "Black", true, LocalDate.parse("21/10/2020", dateFormat));

        System.out.println("Chicken --> " + chicken);

        int id;
        String name;
        String color;
        boolean molting;
        LocalDate bornOnDate;

        //chicken = FileManager.readChickensFromFile("Maruja.json");
        System.out.println("Chicken --> " + chicken);

        // Workshop
        Short minuend;
        short subtrahend;
        short difference;

        minuend = (short) 70000;
        System.out.println(minuend);

        float addend1 = 1.2F;
        float addend2 = 2.4F;
        float sum;
        sum = addend1 + addend2;
        System.out.println(sum);

        char vowell = '5';
        subtrahend = (short) vowell;
        difference = (short) (minuend - subtrahend);
        System.out.println(subtrahend);

        System.out.println(difference);

        // Input data
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter chicken ID: ");
        id = scanner.nextInt();
        scanner.nextLine();

        System.out.print("Enter chicken name: ");
        name = scanner.nextLine();

        System.out.print("Enter chicken color: ");
        color = scanner.nextLine();

        System.out.print("Is the chicken molting? (true/false): ");
        molting = scanner.nextBoolean();
        scanner.nextLine();

        System.out.print("Enter chicken birth date (dd/MM/yyyy): ");
        String bornOnDateString = scanner.nextLine();
        bornOnDate = LocalDate.parse(bornOnDateString, dateFormat);

        chicken = new Chicken(id, name, color, molting, bornOnDate);
        System.out.println("Chicken --> " + chicken);

        FileManager.save(chicken.toString(), "chickens", "json");
    }
}
